pkg/collector/container_test.go

```diff
@@ -458,3 +458,27 @@ service:
 	assert.Equal(t, int32(13133), c.LivenessProbe.HTTPGet.Port.IntVal)
 	assert.Equal(t, "", c.LivenessProbe.HTTPGet.Host)
 }
+
+func TestContainerLifecycle(t *testing.T) {
+	// prepare
+	otelcol := v1alpha1.OpenTelemetryCollector{
+		Spec: v1alpha1.OpenTelemetryCollectorSpec{
+			Lifecycle: &corev1.Lifecycle{
+				PostStart: &corev1.LifecycleHandler{
+					Exec: &corev1.ExecAction{Command: []string{"sh", "sleep 100"}},
+				},
+				PreStop: &corev1.LifecycleHandler{
+					Exec: &corev1.ExecAction{Command: []string{"sh", "sleep 300"}},
+				},
+			},
+		},
+	}
+	cfg := config.New()
+
+	// test
+	c := Container(cfg, logger, otelcol, true)
+
+	// verify
+	assert.Equal(t, []string{"sh", "sleep 100"}, c.Lifecycle.PostStart.Exec.Command)
+	assert.Equal(t, []string{"sh", "sleep 300"}, c.Lifecycle.PreStop.Exec.Command)
```

Comment:
Can we assert that `c`'s entire `Lifecycle` field is equal to an expected `Lifecycle` struct?  I think that ends up being a more precise check.